specification/databricks/resource-manager/Microsoft.Databricks/preview/2025-08-01-preview/databricks.json

```diff
@@ -0,0 +1,1989 @@
+        },
+        "encryption": {
+          "$ref": "#/definitions/WorkspaceEncryptionParameter",
+          "description": "Contains the encryption details for Customer-Managed Key (CMK) enabled workspace."
+        },
+        "requireInfrastructureEncryption": {
+          "$ref": "#/definitions/WorkspaceCustomBooleanParameter",
+          "description": "A boolean indicating whether or not the DBFS root file system will be enabled with secondary layer of encryption with platform managed keys for data at rest."
+        },
+        "storageAccountName": {
+          "$ref": "#/definitions/WorkspaceCustomStringParameter",
+          "description": "Default DBFS storage account name."
+        },
+        "storageAccountSkuName": {
+          "$ref": "#/definitions/WorkspaceCustomStringParameter",
+          "description": "Storage account SKU name, ex: Standard_GRS, Standard_LRS. Refer https://aka.ms/storageskus for valid inputs."
+        },
+        "resourceTags": {
+          "$ref": "#/definitions/WorkspaceCustomObjectParameter",
+          "readOnly": true,
+          "description": "Tags applied to resources under Managed resource group. These can be updated by updating tags at workspace level."
+        }
+      },
+      "required": [
+        "customVirtualNetworkId",
+        "customPublicSubnetName",
+        "customPrivateSubnetName"
+      ],
+      "description": "Custom Parameters used for Cluster Creation."
+    },
+    "CreatedDateTime": {
+      "type": "string",
+      "format": "date-time",
+      "description": "The date and time stamp when the workspace was created.",
+      "readOnly": true
+    },
+    "CreatedBy": {
+      "properties": {
+        "oid": {
+          "type": "string",
+          "format": "uuid",
+          "description": "The Object ID that created the workspace.",
+          "readOnly": true
+        },
+        "puid": {
+          "type": "string",
+          "description": "The Personal Object ID corresponding to the object ID above",
+          "readOnly": true
+        },
+        "applicationId": {
+          "type": "string",
+          "format": "uuid",
+          "description": "The application ID of the application that initiated the creation of the workspace. For example, Azure Portal.",
+          "readOnly": true
+        }
+      },
+      "description": "Provides details of the entity that created/updated the workspace."
+    },
+    "WorkspaceEncryptionParameter": {
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/WorkspaceCustomParameterType",
+          "description": "The type of variable that this is"
+        },
+        "value": {
+          "$ref": "#/definitions/Encryption",
+          "description": "The value which should be used for this field."
+        }
+      },
+      "description": "The object that contains details of encryption used on the workspace."
+    },
+    "Encryption": {
+      "properties": {
+        "keySource": {
```

Comment:
>keySource [](http://example.com/codeflow?start=9&length=9)

Just wondering, is keySource required, or optional?

It seems like the way you're documenting as 'Optional', and 'defaults to Default', it defaults to getting turned off any time someone doesn't specify they want it turned on!

Reply:
Maybe 'EncryptionDisabled' or 'None' would be clearer than keysource = 'Default' then?